Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2282,6 +2282,13 @@ static Vector<FloatRect> absoluteRectsForRangeInText(const SimpleRange& range, T
         return { };
 
     auto offsetRange = characterDataOffsetRange(range, node);
+    // Move to surrogate pair start for Range start and past surrogate pair end for Range end in case the trailing surrogate is indexed.
+    ASSERT(offsetRange.start < node.data().length());
```

Comment:
I don't think this assertion makes sense. It's totally possible for the start to be the end of a text node:
e.g.
```js
document.body.innerHTML = 'hello';
r = document.createRange();
r.setStart(document.body.firstChild, 5);
```

Reply:
Thanks for pointing that out.